jira_sync/main.py

```diff
@@ -169,9 +169,9 @@ def sync_tickets(config_file: str, dry_run: bool):
             # Only move to new state from status we know
             if not (
                 repo_issue.status == IssueStatus.new
-                and jira_issue.fields.status.name not in statuses
+                and jira_issue.fields.status.name not in statuses.values()
```

Comment:
… but rather do this instead:

```suggestion
                and jira_issue.fields.status.name not in statuses.model_fields
```

Reply:
It doesn't return list, but this `{'new': FieldInfo(annotation=str, required=True), 'assigned': FieldInfo(annotation=str, required=True), 'blocked': FieldInfo(annotation=str, required=True), 'closed': FieldInfo(annotation=str, required=True)}`, so not really usable for me. And it doesn't even have the values of the fields, just the definitions.

Reply:
My original mistake begins with that `statuses` is a Pydantic model and continues with that we’re interested in the values, not the keys. Here’s how this could be fixed (beginning around line 58):

```
...
    statuses = jira_config.statuses
    status_strings = statuses.model_dump().values()
...
            if not (
                repo_issue.status == IssueStatus.new
                and jira_issue.fields.status.name not in status_strings
            ):
...
```

Reply:
Needed to wrap the `status_strings` in list, otherwise it returns view.

Reply:
Would have worked nevertheless 😀.